src/Arena.js

```diff
@@ -1,554 +1,555 @@
+  }));
 
   // Setting skills for use in combat. skillArray will contain all usable skills by gladiator at its current level
-  allSkills.map((skill) => {
-    if (skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive){
-      skillArray.push(skill)
-    }
-    return skillArray
-  });
-
-/*---------------------STAT DECLARATIONS-------------------------*/
-    let opponentHP = opponent.hp + battleState.opponentHealth
-    let currentHP = myGladiator.hp + battleState.playerHealth
-    let myHit
-    let oppHit
-    let atkRoll
-    let critRoll
-    let passiveEffect
-    let bleedTick
-    let myAC = myGladiator.ac
-    let critChance = myGladiator.critChance
-    let dmgBonus = myGladiator.dmgBonus
-    let toHitBonus = myGladiator.toHitBonus
-    let oppToHitBonus = opponent.toHitBonus
-    let blockValue = myGladiator.blockValue
-
-//Applies Dex and STR bonuses to hit and damage
-    if (myGladiator.dex > 15){
-      toHitBonus += (myGladiator.dex - 15)
-    }
+  const skillArray = allSkills.filter(
+    (skill) => skill.gladiator === myGladiator.name && myGladiator.level >= skill.lvlUnlock && !skill.passive
+  );
 
-    if (myGladiator.str > 15){
-      dmgBonus += (myGladiator.str - 15)
-    }
+  const [gladiator] = useState(new Gladiator(myGladiator, activeBuffs, dispatch));
+  const [chosenOpponent] = useState(new Opponent(opponent));
 
-//Applies passive skill at level 4
+  const [riposteAnimation, setRiposteAnimation] = useState(null);
+  const [attackAnimation, setAttackAnimation] = useState({enabled: false});
 
-    if (myGladiator.level >= 4){
-      myGladiator.passiveUnlocked = true
-      passiveEffect = myGladiator.passiveEffect
-    }
 
- //Each gladiator has a different bleed tick damage
-    switch (myGladiator.name){
-      case "Dimachaerus":
-        bleedTick = 1
-        break
-      case "Secutor":
-        bleedTick = Math.ceil(dmgBonus/2)
-        break
-      case "Retiarius":
-        bleedTick = Math.ceil(dmgBonus/2) + 3
-        break
-      default:
-        break
-    }
+  let myHit = 0;
 
-/*---------------------PASSIVES, BUFFS AND BUFF DURATIONS---------------------------*/
+  /*---------------------PASSIVES, BUFFS AND BUFF DURATIONS---------------------------*/
 
-  //Riposte function will activate during opponent turn (on successful block)
-  async function riposteAttack() {
-    await sleep(200)
-    let anim = document.createElement("img")
-    myHit = diceRoll(myGladiator.maxDmg) + dmgBonus + toHitBonus;
-    dispatch(riposte(myHit))
-    animatePlayerAttack(myGladiator.name)
-    displayDmgNumber(myHit, "skill-number")
-    anim.className = 'skill-animation'
-    anim.src= require(`./assets/images/animations/riposte.gif`)
-    document.getElementsByClassName('center-screen')[0].appendChild(anim)
-    setTimeout(() => {  anim.parentNode.removeChild(anim); }, 500)
+  function animateAttack(dualWield){
+    setAttackAnimation({enabled: true, dualWield});
```

Comment:
Here's the alternative to doing everything with vanilla dom mutation. You can set a piece of react state to a value here and then down in the JSX, check what that value is for that render cycle and then do something conditionally